onnxruntime/core/providers/cuda/cuda_graph.cc

```diff
@@ -9,15 +9,42 @@
 
 namespace onnxruntime {
 
-CUDAGraph::CUDAGraph(cudaStream_t stream) : stream_(stream) {
+CudaGraphSet::~CudaGraphSet() {
+  Clear();
 }
 
-void CUDAGraph::SetStream(cudaStream_t stream) {
+void CudaGraphSet::Clear() {
+  for (auto& it : cuda_graphs_) {
+    CUDA_CALL_THROW(cudaGraphExecDestroy(it.second));
```

Comment:
Nit: Technically, this is called from within a destructor and it may be best if destructors didn;t throw. But this can be addressed later.

Reply:
right, using CUDA APIs in destructor is not a good practice. but considering https://github.com/pytorch/pytorch/blob/main/aten/src/ATen/cuda/CUDAGraph.cpp#L310-L328, a nice solution may require user to do this manually with provided API which seems not user friendly..